lib/rules/no-unused-vars.js

```diff
@@ -624,10 +624,26 @@ module.exports = {
 
                     // Report the first declaration.
                     if (unusedVar.defs.length > 0) {
+                        let writeReferences;
+
+                        if (unusedVar.references.length > 0) {
+                            writeReferences = unusedVar.references.filter(ref => ref.isWrite());
+                        }
+
+                        let referenceToReport;
+
+                        if (unusedVar.references.length) {
+
+                            // report last write reference, https://github.com/eslint/eslint/issues/14324
+                            if (Array.isArray(writeReferences) && writeReferences.length) {
+                                referenceToReport = writeReferences[writeReferences.length - 1];
+                            } else {
+                                referenceToReport = unusedVar.references[unusedVar.references.length - 1];
+                            }
+                        }
+
                         context.report({
-                            node: unusedVar.references.length ? unusedVar.references[
-                                unusedVar.references.length - 1
-                            ].identifier : unusedVar.identifiers[0],
+                            node: unusedVar.references.length ? referenceToReport.identifier : unusedVar.identifiers[0],
```

Comment:
```suggestion
                            node: referenceToReport ? referenceToReport.identifier : unusedVar.identifiers[0],
```

To avoid duplicating the logic on whether or not a reference should be reported instead of the definition.

Reply:
Updated 👍🏻